ashley/src/main/kotlin/ktx/ashley/engine/engines.kt

```diff
@@ -0,0 +1,52 @@
+package ktx.ashley.engine
+
+import com.badlogic.ashley.core.Component
+import com.badlogic.ashley.core.Engine
+import com.badlogic.ashley.core.Entity
+import ktx.ashley.AshleyDsl
+
+/**
+ * Creates an instance of the component [T] and adds it to this [Entity].
+ *
+ * @param T the [Component] type to get or create
+ * @param configure inlined function with [T] as the receiver to allow additional configuration of the [Component]
+ * @return the created [Component]
+ */
+inline fun <reified T: Component> Entity.with(configure: (@AshleyDsl T).() -> Unit): T {
+  return with<T>().also(configure)
+}
+
+/**
+ * Creates a pooled instance of the component [T] and adds it to this [Entity].
+ *
+ * @param T the [Component] type to get or create
+ * @return the pooled [Component]
+ */
+inline fun <reified T: Component> Entity.with(): T {
+  val component = T::class.java.newInstance()
```

Comment:
Yeah, I think that using plain constructors is preferable. Especially since Kotlin already provides nice syntax for that. You could add unary plus operator to `Component` in context of `Entity`, so you could build your entities like that:

```Kotlin
entity {
  +Transform(
      x = 4f,
      y = 5f
  )
}
```

If you don't think it's sensible to use `Engine` rather than `PooledEngine` directly, you can remove the utilities (at least the conflicting ones, as similar names might cause problems).